Control/Monad/Reader/CPS.hs

```diff
@@ -1,15 +1,18 @@
-{-# LANGUAGE Trustworthy, Rank2Types, FlexibleInstances, FlexibleContexts, MultiParamTypeClasses #-}
+{-# LANGUAGE Trustworthy, Rank2Types, FlexibleInstances, FlexibleContexts, MultiParamTypeClasses, UndecidableInstances #-}
 module Control.Monad.Reader.CPS (ReaderT(..)
     , runReaderT
     , mapReaderT
     , Reader
     , runReader
     , module Control.Monad.Reader.Class) where
 import Control.Monad.Reader.Class
+import Control.Monad.State.Class
 import Control.Applicative
 import Control.Monad.Identity
 import Control.Monad.Trans
-import Unsafe.Coerce
```

Comment:
I don't see any obvious benefit to any of the uses of `unsafeCoerce` in this package, so I've pulled them out. Is there some particular reason you want them?